agent/api/task/task.go

```diff
@@ -1052,7 +1052,7 @@ func (task *Task) initializeASMAuthResource(credentialsManager credentials.Manag
 		if container.ShouldPullWithASMAuth() {
 			container.BuildResourceDependency(asmAuthResource.GetName(),
 				resourcestatus.ResourceStatus(asmauth.ASMAuthStatusCreated),
-				apicontainerstatus.ContainerPulled)
+				apicontainerstatus.ContainerManifestPulled)
```

Comment:
Maybe both `ContainerManifestPulled` and `ContainerPulled` should depend on `ASMAuthStatusCreated` resource status? Do resolved ASM auth creds ever expire?

Reply:
Looks like there is no expiration
https://github.com/aws/amazon-ecs-agent/blob/ff5590b7c5242169eaae234eff2fe761b516042c/agent/taskresource/asmauth/asmauth.go#L386-L410